colossalai/zero/init_ctx/init_context.py

```diff
@@ -82,6 +83,31 @@ def _post_context_exec(self):
         pass
 
 
+class ZeroContextConfig(object):
+
+    def __init__(self,
+                 previous_config=None,
+                 convert_fp16: bool = False,
```

Comment:
is convert_fp16 useful?
@ver217 has removed it from ZeroInitCtx

Reply:
We should set convert_fp16. Otherwise, the weight of gate will be converted to fp16.
@ver217 assumes all parameters should become fp16. Here, we need to have fp32 parameter.

Reply:
Add comments for this class?
some args make developer confusing...

Reply:
Can we just handle gate by ourself? E.g. we can overwrite the `half()` method. Or we can use a flag to indicate some parameters must be fp32.

Reply:
I don't think letting users to control the precision is a good decision.